src/app/modules/twilio/twilio.controller.ts

```diff
@@ -54,6 +54,12 @@ export const twilioSmsUpdates: ControllerHandler<
    * Example: https://www.twilio.com/docs/usage/webhooks/sms-webhooks.
    */
 
+  // Extract public sender's ip address which was passed to twilio as a query param in the status callback
+  const url = req.url.split('?')
+  const paramsString = url.length > 1 ? url[1] : ''
+  const queryParams = new URLSearchParams(paramsString)
+  const senderIp = queryParams.get('senderIp')
```

Comment:
according to the [docs](https://nodejs.org/api/url.html#urlsearchparamsgetname) this can be null right? do we need to deal with that case?

Reply:
yes - it will be null if there is no senderIp (e.g. if it is a bounce sms notification from our server). thereafter we pass it to the logger